src/main/java/org/traccar/timedistancematrix/JsonTimeDistance.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016 - 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2016 - 2017 Andrey Kunitsyn (andrey@traccar.org)
+ * Copyright 2019 - Rafael Miquelino (rafaelmiquelino@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.timedistancematrix;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.traccar.Context;
+
+import javax.json.JsonObject;
+import javax.ws.rs.ClientErrorException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class JsonTimeDistance implements TimeDistanceMatrix {
+
+    static final Logger LOGGER = LoggerFactory.getLogger(JsonTimeDistance.class);
+
+    @Override
+    public TimeDistanceResponse getTimeDistanceMatrix(List<List<Double>> sourceLocations,
+            List<Double> destinationLocation) {
+
+        List<List<Double>> locations = new ArrayList<>(sourceLocations);
+        locations.add(destinationLocation);
+
+        List<Integer> sourceIndexes = new ArrayList<>();
+        for (int i = 0; i < (locations.size() - 1); i++) {
+            sourceIndexes.add(i);
+        }
+
+        List<Integer> destinationIndexes = new ArrayList<>();
+        destinationIndexes.add(locations.size() - 1);
+
+        List<String> metrics = new ArrayList<>();
+        metrics.add("distance");
+        metrics.add("duration");
+
+        TimeDistanceRequest timeDistanceRequest = new TimeDistanceRequest();
+        timeDistanceRequest.setLocations(locations);
+        timeDistanceRequest.setSources(sourceIndexes);
+        timeDistanceRequest.setDestinations(destinationIndexes);
+        timeDistanceRequest.setMetrics(metrics);
+
+        JsonObject resultJson = null;
+        try {
+            resultJson = getTimeDistanceResponse(timeDistanceRequest);
+        } catch (ClientErrorException e) {
+            LOGGER.warn("Time distance network error", e);
+        }
+
+        TimeDistanceResponse result = null;
+        try {
+            result = Context.getObjectMapper().readerFor(TimeDistanceResponse.class)
```

Comment:
Are you expecting all providers to return same format? I feel like that's too optimistic.